trunk/src/app/srs_app_source.cpp

```diff
@@ -1993,7 +1980,7 @@ bool SrsLiveSource::publisher_is_idle_for(srs_utime_t timeout)
     return false;
 }
 
-srs_error_t SrsLiveSource::initialize(SrsRequest* r, ISrsLiveSourceHandler* h)
+srs_error_t SrsLiveSource::initialize(SrsSharedPtr<SrsLiveSource> wrapper, SrsRequest* r, ISrsLiveSourceHandler* h)
```

Comment:
this method and every initialize embedded this is method can be declared to just accept a `SrsSharedPtr` reference. It will save a lot of `SrsSharedPtr` construction and destruction operations.

Reply:
I have claimed for many times: If you want to improve performance, please provide benchmark data. Most of performance improvement does not work. Most performance improvement work does not work as expectation. Never fix none-exists problem.

For this case, `initialize` is not hot path, so there is no performance bottleneck. We should never use reference if do not need it.

Won't fix.